### DISCUSSION TITLE
Compute renderer, will Android have performance improvements?

### DISCUSSION THREAD

**Discussion by ghost** [rafaelvcaetano/melonDS-android#328] [Q&A]
[Generic/RSDuck wrote a post explaining the new compute renderer to allow for full speed emulation of 3D games on platforms like the Switch that has weak CPU but good GPU.](http://melonds.kuribo64.net/comments.php?id=143)

I know the OpenGL renderer isn't ready on Android but will this have any performance improvements for Android or will it be about the same as it is now?

MelonDS runs perfectly on my OnePlus 7T Pro but that has a Snapdragon 855+ so I don't know how performance will scale for the 7XX and 6XX series of SoCs. I'm looking forward for upscaling, even just a x2 resolution makes 3D elements in games look so much better and texture replacement when that's implemented.

**Comment by rafaelvcaetano** [comment]
I'm not sure how easy it will be to port that renderer to Android. I've never used compute shaders so it will be quite interesting to see what it's capable of. As such, I can't tell you right now if it will bring performance improvements or if it will even be supported, but I can guarantee you that I'll mess around with it to see what's possible.

**Reply by ghost** [reply]
Cool, I understand that OpenGLES isn't the same as OpenGL and that Android is missing things that are available on PC so porting is complicated. Personally I find it frustrating that there has never been attempts to improve Android on this front, I don't think the hardware is the issue but Google keeps breaking things in Android that make life harder like scoped storage on Android 11.

**Reply by RayyanAnsari** [reply]
The compute renderer might have performance improvements on devices with better GPUs.
A better option would be to port the OpenGL renderer to GLES becwuse it will be faster, but maybe less accurate.
I have attempted this here: https://github.com/Arisotura/melonDS/tree/opengl-es-renderer
Due to my lack of knowledge on this, there are issues with 3D.
I would appreciate any help!

**Comment by Chieh-Huang** [comment]
Yes, render can improve performance.
https://youtu.be/P_I8an8jXuM
But render are difficult to learning, Vulkan especially.
